repl/src/main/scala/com/cloudera/livy/repl/Session.scala

```diff
@@ -258,21 +258,17 @@ class Session(
     executeCode(statementId, cmd)
   }
 
-  private def cleanOldStatements(): Unit = {
+  private def cleanOldStatements(): Unit = _statements.synchronized {
```

Comment:
Since we are using LinkedHashMap, I can we can implement [removeEldestEntry()](https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html#removeEldestEntry-java.util.Map.Entry-).